File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/endpoint/dto/UserDetailDto.java
package at.ac.tuwien.sepr.groupphase.backend.endpoint.dto;

import at.ac.tuwien.sepr.groupphase.backend.enums.UserRole;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode
@AllArgsConstructor
@ToString
public class UserDetailDto {

    private Long id;

    private String email;

    private String firstName;

    private String lastName;

    private Boolean isLocked;

    private UserRole role;

    private UserLocationDto location;
}

File: frontend/src/app/dtos/user-detail-dto.ts
import { UserLocationDto } from './user-location-dto';
import { UserRole } from '../enums/userRole';

export interface UserDetailDto {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  isLocked: boolean;
  role: UserRole;
  location?: UserLocationDto;
}
